vyper/context/namespace.py

```diff
@@ -94,3 +96,93 @@ def get_namespace():
     except NameError:
         _namespace = Namespace()
         return _namespace
+
+
+def validate_identifier(attr):
+    namespace = get_namespace()
+    if attr in namespace and attr not in [x for i in namespace._scopes for x in i]:
+        raise NamespaceCollision(f"Cannot assign to '{attr}', it is a builtin")
+    if attr.lower() in RESERVED_KEYWORDS or attr.upper() in OPCODES:
+        raise StructureException(f"'{attr}' is a reserved keyword")
+    if not re.match("^[_a-zA-Z][a-zA-Z0-9_]*$", attr):
+        raise StructureException(f"'{attr}' contains invalid character(s)")
+
+
+# Cannot be used for variable or member naming
+RESERVED_KEYWORDS = {
```

Comment:
it would be lovely if we can somehow pull this from our stuff dyanmically, instead of statically maintaining this alongside changes to the compiler

Reply:
Definitely worth implementing.